exporter/elasticsearchexporter/model.go

```diff
@@ -596,6 +740,37 @@ func metricHash(timestamp pcommon.Timestamp, attributes pcommon.Map) uint32 {
 	return hasher.Sum32()
 }
 
+func metricOTelHash(dp dataPoint, scopeAttrs pcommon.Map, unit string) uint32 {
+	hasher := fnv.New32a()
+
+	timestampBuf := make([]byte, 8)
+	binary.LittleEndian.PutUint64(timestampBuf, uint64(dp.Timestamp()))
+	hasher.Write(timestampBuf)
+
+	binary.LittleEndian.PutUint64(timestampBuf, uint64(dp.StartTimestamp()))
+	hasher.Write(timestampBuf)
+
+	hasher.Write([]byte(unit))
+
+	mapHashExcludeDataStreamAttr(hasher, scopeAttrs)
+	mapHashExcludeDataStreamAttr(hasher, dp.Attributes())
```

Comment:
We're excluding them because docs are already grouped by resource, and `data_stream.*` is set in the resource attributes by the document router - right? Maybe worth another comment here?

Reply:
I'll add a comment. It is because e.g. if both scope DS dataset and record level DS dataset are present, scope DS dataset will effectively be ignored, and will not affect the resulting document, as the record level one will be used. We'll risk duplicates if we factor it in during hashing.

Reply:
On the other hand, documents are already passed into the function is already "sharded" by index name, which is also a reason why it isn't needed in hashing.

Reply:
thanks for making me think more about it. ECS mode should use this mapHashExcludeDataStreamAttr as well.